### PR TITLE
use load instead of convert

### PR DESCRIPTION
**Pull request by Mogztter**
#### What I Did

- Add tests on `Import.Image.getCurrentImagesDir`
- Change the signature of `getCurrentImagesDir` to make it easier to test
- Fix the regular expression that checks if the string starts with a protocol: `/^(?:[a-z]+:)?\/\//i` (remote)
- Use the `load` function in `attributeReferenceProvider.ts`, `tableOfContentsProvider.ts` and `image-paste.ts` since we only need an Asciidoctor.Document (instead of the parse/convert)

In a following pull request, I will probably remove a layer of indirection since we don't really need both `AsciidocEngine` and `AsciidocParser`.
In other words, `AsciidocEngine` has little to no value.

### CODE REVIEW

**Comment by danyill** [comment]
> Add tests on Import.Image.getCurrentImagesDir

I'm so happy to see some tests! :grinning: 

> In a following pull request, I will probably remove a layer of indirection since we don't really need both AsciidocEngine and AsciidocParser. In other words, AsciidocEngine has little to no value.

One of the problems with the existing approach is sometimes we've been directly calling the convertUsingJavascript and there are times we've been doing it through the AsciidocEngine which is instantiated on startup.
They're not the same and there are undesirable side effects on error messages and other things so I think this would be an improvement.

**Comment by danyill** [comment]
> Add tests on Import.Image.getCurrentImagesDir

I'm so happy to see some tests! :grinning: 

> In a following pull request, I will probably remove a layer of indirection since we don't really need both AsciidocEngine and AsciidocParser. In other words, AsciidocEngine has little to no value.

One of the problems with the existing approach is sometimes we've been directly calling the convertUsingJavascript and there are times we've been doing it through the AsciidocEngine which is instantiated on startup.
They're not the same and there are undesirable side effects on error messages and other things so I think this would be an improvement.
